retrievalmarket/impl/provider_environments.go

```diff
@@ -30,17 +30,50 @@ type providerValidationEnvironment struct {
 	p *Provider
 }
 
-func (pve *providerValidationEnvironment) GetPiece(c cid.Cid, pieceCID *cid.Cid) (piecestore.PieceInfo, error) {
+func (pve *providerValidationEnvironment) GetAsk(ctx context.Context, payloadCid cid.Cid, pieceCid *cid.Cid,
```

Comment:
It seems like all the code in this function is shared with some code in Provider.HandleQueryStream()
Can we refactor to keep the common code in one place?